lfs/gitscanner_log.go

```diff
@@ -279,51 +280,70 @@ func (s *logScanner) finishLastPointer() *WrappedPointer {
 // There can be multiple diffs per commit (multiple binaries)
 // Also when a binary is changed the diff will include a '-' line for the old SHA
 func (s *logScanner) scan() (*WrappedPointer, bool) {
-	for s.s.Scan() {
-		line := s.s.Text()
-
-		if match := s.commitHeaderRegex.FindStringSubmatch(line); match != nil {
-			// Currently we're not pulling out commit groupings, but could if we wanted
-			// This just acts as a delimiter for finishing a multiline pointer
-			if p := s.finishLastPointer(); p != nil {
-				return p, true
-			}
-		} else if match := s.fileHeaderRegex.FindStringSubmatch(line); match != nil {
-			// Finding a regular file header
-			p := s.finishLastPointer()
+	long := false
+	for {
+		bytes, prefix, err := s.r.ReadLine()
 
-			// Pertinent file name depends on whether we're listening to additions or removals
-			if s.dir == LogDiffAdditions {
-				s.setFilename(match[2])
+		if err != nil {
+			if err == io.EOF {
+				break
 			} else {
-				s.setFilename(match[1])
+				s.err = err
+				return nil, false
 			}
+		}
 
-			if p != nil {
-				return p, true
-			}
-		} else if match := s.fileMergeHeaderRegex.FindStringSubmatch(line); match != nil {
-			// Git merge file header is a little different, only one file
-			p := s.finishLastPointer()
+		if prefix {
+			long = true
+		} else if long {
+			long = false
+			tracerx.Printf("Skipping long line ending with: %v", string(bytes))
+		} else {
+			line := string(bytes)
+
+			if match := s.commitHeaderRegex.FindStringSubmatch(line); match != nil {
+				// Currently we're not pulling out commit groupings, but could if we wanted
+				// This just acts as a delimiter for finishing a multiline pointer
+				if p := s.finishLastPointer(); p != nil {
+					return p, true
+				}
+			} else if match := s.fileHeaderRegex.FindStringSubmatch(line); match != nil {
+				// Finding a regular file header
+				p := s.finishLastPointer()
+
+				// Pertinent file name depends on whether we're listening to additions or removals
+				if s.dir == LogDiffAdditions {
+					s.setFilename(match[2])
+				} else {
+					s.setFilename(match[1])
+				}
 
-			s.setFilename(match[1])
+				if p != nil {
+					return p, true
+				}
+			} else if match := s.fileMergeHeaderRegex.FindStringSubmatch(line); match != nil {
```

Comment:
This is all good stuff.  I think it's probably outside the scope of this PR, but it would certainly be interesting to do some profiling.

Reply:
After a night's reflection, I actually think both of these changes should go into a separate PR, not just the latter one, so I'm going to mark this resolved and write up a separate one when I get a chance (with some additional testing, since these are potentially serious changes in functionality).  Thanks for listening, though!  :-)